packages/amplify_authenticator/example/integration_test/sign_in_mfa_test.dart

```diff
@@ -0,0 +1,180 @@
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+// This test follows the Amplify UI feature "sign-in-with-username"
+// https://github.com/aws-amplify/amplify-ui/blob/main/packages/e2e/features/ui/components/authenticator/sign-up-with-username.feature
+import 'dart:async';
+
+import 'package:amplify_api/amplify_api.dart';
+import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+import 'package:amplify_authenticator/amplify_authenticator.dart';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+import 'config.dart';
+import 'pages/confirm_sign_in_page.dart';
+import 'pages/confirm_sign_up_page.dart';
+import 'pages/sign_in_page.dart';
+import 'pages/test_utils.dart';
+import 'utils/data_utils.dart';
+import 'utils/mock_data.dart';
+
+void main() {
+  final binding = IntegrationTestWidgetsFlutterBinding.ensureInitialized()
+      as IntegrationTestWidgetsFlutterBinding;
+  // resolves issue on iOS. See: https://github.com/flutter/flutter/issues/89651
+  binding.deferFirstFrame();
+
+  final authenticator = MaterialApp(
+    home: Authenticator(
+        child: const SignOutButton(
+      key: Key('keySignOutButton'),
+    )),
+  );
+
+  group('sign-in-sms-mfa', () {
+    // Given I'm running the example "ui/components/authenticator/sign-in-sms-mfa.feature"
+    setUpAll(() async {
+      await loadConfiguration(
+        'ui/components/authenticator/sign-in-sms-mfa',
+        additionalConfigs: [AmplifyAPI()],
+      );
+    });
+
+    // Scenario: Sign in using a valid phone number and SMS MFA
+    testWidgets('Sign in using a valid phone number and SMS MFA',
+        (tester) async {
+      final phone = generatePhone();
+      final password = generatePassword();
+      await adminCreateUser(phone, password,
+          autoConfirm: true, enableMfa: true, verifyAttributes: true);
+      await loadAuthenticator(tester: tester, authenticator: authenticator);
+      SignInPage signInPage = SignInPage(tester: tester);
+      ConfirmSignInPage confirmSignInPage = ConfirmSignInPage(tester: tester);
+
+      signInPage.expectUserNameIsPresent(usernameLabel: 'Phone Number');
+
+      // When I select my country code
+      await signInPage.selectCountryCode();
+
+      // And I type my "phone number" with status "CONFIRMED"
+      await signInPage.enterUsername(phone);
+
+      // And I type my password
+      await signInPage.enterPassword(password);
+
+      // And I click the "Sign in" button
+      await signInPage.submitSignIn();
+
+      // Then I will be redirected to the confirm sms mfa page
+      await confirmSignInPage.expectConfirmSignInMFAIsPresent();
```

Comment:
I know it's not part of the original - but can we can complete the sign in flow? I suspect they haven't added that since they don't have the infra yet, but we can now.

Reply:
Yeah for sure - I was trying to get this up since it's the baseline (and also to make sure nobody has issues with the AppSync resolvers that would require changes).  I'm fine either merging this as-is or adding the MFA integration. Just wanted to get the PR up.